test/tensors/factorizations.jl

```diff
@@ -329,36 +334,38 @@ for V in spacelist
                 @test isposdef(t - λ * one(t) + 0.1 * one(t))
                 @test !isposdef(t - λ * one(t) - 0.1 * one(t))
 
-                d, v = @constinferred eigh_trunc(t; trunc = truncrank(dim(domain(t)) ÷ 2))
+                d, v = @constinferred eigh_trunc(t; trunc = truncrank(nvals))
                 @test t * v ≈ v * d
-                @test dim(domain(d)) ≤ dim(domain(t)) ÷ 2
+                @test dim(domain(d)) ≤ nvals
             end
         end
 
         @testset "Condition number and rank" begin
             for T in eltypes,
                     t in (
                         rand(T, W, W), rand(T, W, W)',
-                        rand(T, W, V1), rand(T, V1, W),
-                        rand(T, W, V1)', rand(T, V1, W)',
+                        rand(T, W, V4), rand(T, V4, W),
+                        rand(T, W, V4)', rand(T, V4, W)',
                         DiagonalTensorMap(rand(T, reduceddim(V1)), V1),
                     )
 
                 d1, d2 = dim(codomain(t)), dim(domain(t))
-                @test rank(t) == min(d1, d2)
+                r = rank(t)
+                @test r == min(d1, d2)
+                @test eltype(r) == eltype(d1)
```

Comment:
```suggestion
                @test typeof(r) == typeof(d1)
```
This should boil down to the same thing, but `eltype` confused me for a second since this are just scalars and that seems to imply they are matrices